R/data-gather-clean.R

```diff
@@ -4,178 +4,190 @@
-get_submissions <- function(syn, group, statuses,
-                            section_lookup_table, variable_lookup_table) {
+get_submissions <- function(syn, group, statuses, lookup_table) {
   if (is.null(statuses)) {
     return(NULL)
   }
-  submissions <- purrr::map(statuses, function(x) {
-    synapseforms::download_all_and_get_table(
+  sub_list <- purrr::flatten(purrr::map(statuses, function(x) {
+    synapseforms::download_all_submissions_temp(
       syn = syn,
       state_filter = x,
       group = group
     )
-  })
-  if (all(is.null(unlist(submissions)))) {
+  }))
+  if (all(is.null(unlist(sub_list)))) {
     return(NULL)
   }
-  submissions <- submissions  %>%
-    purrr::compact() %>% # Removes NAs
-    purrr::reduce(dplyr::full_join, by = "variables")
-  submissions <- make_clean_table(
-    submissions,
-    section_lookup_table,
-    variable_lookup_table
-  )
-  submissions
-}
 
-#' Make clean submissions table
-#'
-#' Make submissions table clean by fixing experiment
-#' numbering and adding in user-friendly step and label
-#' columns instead of json variable styled section and
-#' variable columns.
-#'
-#' @keywords internal
-#' @inheritParams get_submissions
-#' @inheritParams clean_experiment_variables
-make_clean_table <- function(data, section_lookup_table,
-                             variable_lookup_table) {
-  data <- synapseforms::make_tidier_table(data)
-  data <- dplyr::left_join(data, add_friendly_names(data), by = "form_data_id")
-  data <- data[!is.na(data$response), ]
-  data <- dplyr::left_join(data, section_lookup_table, by = "section")
-  data <- clean_experiment_variables(data)
-  data <- dplyr::left_join(data, variable_lookup_table, by = "variable")
-  # Some labels/steps will be blank if form changes or not mapped
-  na_labels <- which(is.na(data$label))
-  data$label[na_labels] <- data$variable[na_labels]
-  na_steps <- which(is.na(data$step))
-  data$step[na_steps] <- data$section[na_steps]
-  # Only need nice, curated table
-  data <- data[c(
-    "form_data_id",
+  ## Main table creation, along with submission name
+  all_subs <- purrr::map2_dfr(
+    sub_list,
+    names(sub_list),
+    function(filename, data_id) {
+      data <- jsonlite::fromJSON(filename, simplifyVector = FALSE)
+      sub <- purrr::imap_dfr(data, create_table_by_sections, TRUE)
+      sub <- tibble::add_column(sub, form_data_id = data_id)
+      # Form submission name
+      user_name <- sub$response[which(sub$variable == "last_name")]
+      compound_name <- sub$response[which(sub$variable == "compound_name")]
+      sub <- tibble::add_column(
+        sub,
+        submission = glue::glue("{user_name} - {compound_name}")
+      )
+      sub
+    }
+  )
+  ## Remove metadata section
+  metadata_indices <- which(all_subs$section == "metadata")
+  if (length(metadata_indices) > 0) {
+    all_subs <- all_subs[-metadata_indices, ]
+  }
+  ## Get mapped steps and labels appended with experiment number on step
```

Comment:
```suggestion
  ## Add columns 'step' and 'label', which contain user-friendly display names
  ## (including the experiment number) for the sections and variables
```
small change but I think this makes it clearer what is happening in this line